ee/clickhouse/sql/retention/retention.py

```diff
@@ -15,6 +15,34 @@
 ORDER BY base_interval, intervals_from_base
 """
 
+RETENTION_BREAKDOWN_SQL = """
+    SELECT
+        target_event.breakdown_values AS breakdown_values,
+        datediff(
+            %(period)s, 
+            target_event.event_date, 
+            dateTrunc(%(period)s, toDateTime(returning_event.event_date))
+        ) AS intervals_from_base,
+        COUNT(DISTINCT returning_event.target) AS count
+
+    FROM
+        ({returning_event_query}) AS returning_event
+        JOIN ({target_event_query}) target_event
+            ON returning_event.target = target_event.target
+
+    WHERE 
+        dateTrunc(%(period)s, returning_event.event_date) >
+        dateTrunc(%(period)s, target_event.event_date)
+
+    GROUP BY 
+        breakdown_values, 
+        intervals_from_base
+
+    ORDER BY 
+        breakdown_values, 
+        intervals_from_base
+"""
+
```

Comment:
should add a benchmark for this new query and try out some version of this query on production data/metabase. Retention isn't that performant already so we should have some idea of how much more taxing this query is

Reply:
Have added https://github.com/PostHog/posthog/pull/7483 for the non-breakdown query and rebased on top of that. I'll rerun the benchmarks once that is merged and verify that there are no regressions.

I'll add one for the new as well in this PR.

Reply:
I've added a benchmark for breakdown also.